rom/laghos.cpp

```diff
@@ -1576,19 +1617,51 @@ int main(int argc, char *argv[])
             {
                 timeLoopTimer.Stop();
                 samplerTimer.Start();
+
+                double real_pd;
                 if (rom_sample_stages)
                 {
                     std::vector<Vector>& RKStages = ode_solver_samp->GetRKStages();
                     std::vector<double>& RKTime = ode_solver_samp->GetRKTime();
                     MFEM_VERIFY(RKStages.size() == RKStepNumSamples, "Inconsistent number of Runge Kutta stages.");
                     for (int RKidx = 0; RKidx < RKStepNumSamples; ++RKidx)
                     {
-                        sampler->SampleSolution(RKTime[RKidx], last_dt, RKStages[RKidx]);
-                        if (samplerLast) samplerLast->SampleSolution(RKTime[RKidx], dt, RKStages[RKidx]);
+                        if (problem == 7)
+                        {
+                            // 2D Rayleigh-Taylor penetration distance
+                            if (romOptions.indicatorType == penetrationDistance)
+                            {
+                                real_pd = -1.0;
+                                double proc_pd = (pd2_vdof >= 0) ? -RKStages[RKidx](pd2_vdof) : 0.0;
+                                MPI_Reduce(&proc_pd, &real_pd, 1, MPI_DOUBLE, MPI_MAX, 0, MPI_COMM_WORLD);
```

Comment:
Should this be `MPI_Allreduce`? Using `MPI_Reduce` just computes the result `real_pd` on the root process, but isn't it needed on all processes?

Reply:
I believe  `MPI_Reduce` suffices since the use of `real_pd` here is to record the FOM penetration distance in the offline phase, which will be written into a file for merge phase (if it is parametric problem) and online phase (for the determination of window change) and only the master node needs a record.

Reply:
Thanks for clarifying. I see now that it is used only by the root node. It is still not clear to me how the `MFEM_VERIFY` does not fail for processes other than root, which should have `real_pd == -1.0`. Have you tested this in parallel? Also, I think it would be a little better to set `double real_pd = -1.0;` in the declaration of this variable, so it is not uninitialized in some case.

Reply:
I noticed that you changed some `MPI_Reduce` calls to `MPI_Allreduce`, but there are still some left (related to `real_pd`).